Fix possible crash in joy_axis (fixes #4944).

marcelofg55 opened the pull request:
Just adds a missing check to prevent a crash.

dumitru-stama committed:
Fixed a string class bug

marcelofg55 committed:
Fix possible crash in joy_axis (fixes #4944).

Hinsbart commented on main/input_default.cpp:
```diff
@@ -876,6 +876,10 @@ void InputDefault::joy_axis(int p_device, int p_axis, const JoyAxis &p_value) {
 
 	Joypad &joy = joy_names[p_device];
 
+	if (p_axis < 0 || p_axis >= JOY_AXIS_MAX) {
```
I guess this should throw an error.
You can do it with `ERR_FAIL_INDEX(p_axis, JOY_AXIS_MAX)`

marcelofg55 replied:
I've added it and commited the change, and I tried to squash the commits but I think it didn't work well :(

Hinsbart replied:
Ah, you can remove  the `if (p_axis < 0 || p_axis >= JOY_AXIS_MAX) ` and the return statement, `ERR_FAIL_INDEX()` already does all of that for you :)

27thLiz committed:
Fix EditorImportPlugin::get_preset_count

Was missing a const...

karroffel committed:
fixed ptrcall cast for const Ref<T>

Some methods require a const Ref<T> argument,
the ptrcall method wrappers cast `void *` to the
apropriate types. The problem is that there is no `Ref(const T *)`
constructor, but since Ref modifies the refcount of a Reference
anyway there's no point in a const version.

The problem is that with a `const T *` constructor call, the
argument gets converted to Variant first and loses all the
reference information, resulting in a null reference as the
argument to the constructor.

marcelofg55 committed:
Added ERR_FAIL_INDEX for the p_axis error check at joy_axis.
